src/utils/useDefaultDate.ts

```diff
@@ -16,9 +19,15 @@ import { DEFAULT_DATE_FORMAT } from './name-utils';
  */
 export function useDefaultDate(
   availableDatesLookupKey: keyof AvailableDates,
-  layerGroup?: GroupDefinition,
+  layerId?: LayerKey,
 ): number | undefined {
   const dispatch = useDispatch();
+  const selectedLayers = useSelector(layersSelector);
+  // check layer without group or main layer in group
+  const isMainLayer = !selectedLayers
```

Comment:
some of the filtering is a bit duplcative throughout the code.

maybe expose `isMainLayer` as a function  next to `isLayerKey` for example so that you can re-use it